### PR TITLE
fix: APP-1018, fix update chatroom to im

### PR DESCRIPTION
**Pull request by pdarde-daitan**
When the user updates a webhook instance from CHATROOM to IM the list of streams becomes empty
https://perzoinc.atlassian.net/browse/APP-1018

### CODE REVIEW

File: sagas/editInstance.js [hunk 1]
```diff
@@ -3,14 +3,19 @@ import {
...
 } from './apiCalls';
 
 export function* editInstance() {
+  let imStream;
   try {
     const state = yield select();
     yield call(setInstance, state.instance);
-    if (state.instance.streamType === 'CHATROOM') {
+    if (state.instance.streamType === 'IM') {
+      imStream = yield call(createIM);
```
Comment (ecarrenho-daitan) [inline sagas/editInstance.js:17]: Don't we need to handle the error scenario when creating the IM here?

File: sagas/editInstance.js [hunk 2]
```diff
@@ -3,14 +3,19 @@ import {
...
+      state.instance.streams.push(imStream.id);
```
Comment (ecarrenho-daitan) [inline sagas/editInstance.js]: Should this be a push on the list of streams? Or we should replace it?

File: sagas/editInstance.js [hunk 3]
```diff
@@ -3,26 +3,35 @@ import {
...
+            status = yield call(addMembership, state.instance.streams[stream]);
           }
         }
       }
     }
+    if (
+      status.message === 'Member added' ||
+      status.message === 'Member already part of the room or Xpod request will be processed asynchronously'
+    ) {
     yield call(callEditInstance, state);
```
Comment (ecarrenho-daitan) [inline sagas/editInstance.js]: Fix identation.

File: sagas/editInstance.js (hunk 1 above)
Comment (pdarde-daitan) [reply sagas/editInstance.js:17]: Yes, it's done.

**Review by ecarrenho-daitan — APPROVED** [review]
LGTM

**Pull request merged**